src/main/kotlin/interpreter/functions/PyFunction.kt

```diff
@@ -29,9 +29,6 @@ import green.sailor.kython.interpreter.pyobject.PyType
 abstract class PyFunction(type: PyType) : PyObject(type), PyCallable {
     // overridden for method binding
     override fun pyDescriptorGet(parent: PyObject, klass: PyObject): PyObject {
-        if (parent is PyNone) {
-            return this
-        }
-        return PyMethod(this, parent)
+        return takeIf { parent is PyNone } ?: PyMethod(this, parent)
```

Comment:
Extremely against this

Reply:
Why?

Reply:
I like regular if statements over a functional if statement, especially when if is an expression, this feels like functional for the sake of being functional

Reply:
Sure, I'll change it